### ISSUE TITLE
The component with the UUID does not exist in the project's library!

### ISSUE THREAD

**Issue by shakenov-chinga** [LibrePCB/LibrePCB#390]
Hi guys!
I was working on my new project, and decided to try LibrePCB for that purpose. You made quite a progress since last year and I want to thank you for your hard work :+1: .
So, I created several libraries and used it in the project. Then I made little changes in one of its component, updated libraries from within schematics window and got error window with message 
`The component with the UUID "aeb89f16-126b-40b7-84e1-242b0ec7eafd" does not exist in the project's library!`
It does exist, and it is from a different library, where I didn't touch anything.

I started LibrePCB using terminal, here is an output:
~/LibrePCB/stable/bin$ ./librepcb
Load stroke font: "newstroke.bene"
Start loading font "/home/chinga/LibrePCB/stable/share/librepcb/fontobene/newstroke.bene"
[  INFO   ] "LibrePCB 0.1.0 (d7458d3b3e126499902e1a66a0ef889f516a7c97)" (:0)
[  INFO   ] "Qt version: 5.9.1 (compiled against 5.9.1)" (:0)
[  INFO   ] "Resources directory: /home/chinga/LibrePCB/stable/share/librepcb" (:0)
[  INFO   ] "Application settings: /home/chinga/.config/LibrePCB/LibrePCB.ini" (:0)
[DEBUG-MSG] Started network access manager thread. (:0)
[ WARNING ] There was a stale lock on the workspace: "FilePath(/home/chinga/Dropbox/LibrePCB)" (:0)
[DEBUG-MSG] Load workspace settings... (:0)
[  INFO   ] Workspace settings file not found, default settings will be used. (:0)
[DEBUG-MSG] Workspace settings successfully loaded! (:0)
[DEBUG-MSG] Load local workspace library: "connectors.lplib" (:0)
[EXCEPTION] Not a valid URL. (../../../../libs/librepcb/common/fileio/../fileio/sexpression.h:282)
[EXCEPTION] File parse error: Not a valid URL.

File: /home/chinga/Dropbox/LibrePCB/v0.1/libraries/local/connectors.lplib/library.lp
Line,Column: -1,-1
Invalid Content: "" (../../../../libs/librepcb/common/fileio/../fileio/sexpression.h:98)
[ WARNING ] "File parse error: Not a valid URL.\n\nFile: /home/chinga/Dropbox/LibrePCB/v0.1/libraries/local/connectors.lplib/library.lp\nLine,Column: -1,-1\nInvalid Content: \"\"" (:0)
[DEBUG-MSG] Load local workspace library: "projects.lplib" (:0)
[EXCEPTION] Not a valid URL. (../../../../libs/librepcb/common/fileio/../fileio/sexpression.h:282)
[EXCEPTION] File parse error: Not a valid URL.

File: /home/chinga/Dropbox/LibrePCB/v0.1/libraries/local/projects.lplib/library.lp
Line,Column: -1,-1
Invalid Content: "" (../../../../libs/librepcb/common/fileio/../fileio/sexpression.h:98)
[ WARNING ] "File parse error: Not a valid URL.\n\nFile: /home/chinga/Dropbox/LibrePCB/v0.1/libraries/local/projects.lplib/library.lp\nLine,Column: -1,-1\nInvalid Content: \"\"" (:0)
[DEBUG-MSG] Load remote workspace library: "326f091b-b715-44bf-b385-b613cd60d9f3.lplib" (:0)
[DEBUG-MSG] Load remote workspace library: "5d00d6ec-ce1e-4a1c-b5fb-78d1d41bdde7.lplib" (:0)
[DEBUG-MSG] Load remote workspace library: "6ccc516c-21b7-4cd5-9cf2-7a04cfa361c6.lplib" (:0)
[DEBUG-MSG] Load remote workspace library: "a9ddf0c6-9b1c-4730-b300-01b4f192ad40.lplib" (:0)
[DEBUG-MSG] Load remote workspace library: "b404d66b-62ea-4e55-97fd-bbc1be7c297a.lplib" (:0)
[DEBUG-MSG] Load workspace library database... (:0)
[DEBUG-MSG] Workspace library database successfully loaded! (:0)
[DEBUG-MSG] Successfully loaded font "/home/chinga/LibrePCB/stable/share/librepcb/fontobene/newstroke.bene" with 2573 glyphs (:0)
[DEBUG-MSG] open project: "/home/chinga/Dropbox/LibrePCB/projects/Project/Project.lpp" (:0)
[DEBUG-MSG] Load stroke font: "newstroke.bene" (:0)
[DEBUG-MSG] Start loading font "/home/chinga/Dropbox/LibrePCB/projects/Project/resources/fontobene/newstroke.bene" (:0)
[DEBUG-MSG] load project metadata... (:0)
[DEBUG-MSG] metadata successfully loaded! (:0)
[DEBUG-MSG] load settings... (:0)
[DEBUG-MSG] settings successfully loaded! (:0)
[DEBUG-MSG] load project library... (:0)
[DEBUG-MSG] successfully loaded 0 symbols (:0)
[DEBUG-MSG] successfully loaded 0 packages (:0)
[DEBUG-MSG] successfully loaded 0 components (:0)
[DEBUG-MSG] successfully loaded 0 devices (:0)
[DEBUG-MSG] project library successfully loaded! (:0)
[DEBUG-MSG] load circuit... (:0)
[EXCEPTION] The component with the UUID "aeb89f16-126b-40b7-84e1-242b0ec7eafd" does not exist in the project's library! (../../../../libs/librepcb/project/circuit/componentinstance.cpp:66)

I work on two different PCs with Linux Mint 18.3 and Debian 9.6. Both show the same output.
I can share my workspace later, if it is needed.
Thanks in advance!

**Comment by ubruhin** [comment]
> Hi guys!
I was working on my new project, and decided to try LibrePCB for that purpose. You made quite a progress since last year and I want to thank you for your hard work +1 .

Hi @shakenov-chinga, you're welcome! And thanks for trying LibrePCB :smiley: 

> Then I made little changes in one of its component, updated libraries from within schematics window and got error window with message
The component with the UUID "aeb89f16-126b-40b7-84e1-242b0ec7eafd" does not exist in the project's library!

Did this error only appear in the project library updater and the project is still working properly (except that it can't be updated)? Or is it now permanently broken, so you can't open the project anymore?

>  [ WARNING ] There was a stale lock on the workspace: "FilePath(/home/chinga/Dropbox/LibrePCB)" (:0)

Did LibrePCB maybe crash during the library update or so?

**Comment by shakenov-chinga** [comment]
Hi,

> Did this error only appear in the project library updater and the project is still working properly (except that it can't be updated)? Or is it now permanently broken, so you can't open the project anymore?

Error appeared in the project library updater and project was permanently broken.

> Did LibrePCB maybe crash during the library update or so?
No, LibrePCB didn't crash at all. The library updater showed errors after all updates, but didn't crash. Unfortunately, I don't remember what kind of errors was in the library updater. Control Panel was still running and I were able to create and edit other projects, but unable to open my project.

I think I fixed it now. I compared my project with some other dummy project and figured out that library directory in my project folder was empty. I copied cmp, dev, pkg and sym folders from remote and local library lplib folders. I think I copied some unnecessary libraries, because project opens now with big delay.

I weren't able to recreate this error again.  I'll give an update if I have this error again.
Thank you very much!

**Comment by shakenov-chinga** [comment]
Too many library instances in the project freezes my schematic and board editor windows. Therefore, I had to delete all and manually copy needed parts from cmp, dev, pkg and sym folders one by one. It wasn't easy, since folders are named by uuid values, but still works :smiley:!

It would be good to implement library checking feature, which will add needed parts from remote and local libraries and delete unused ones. :smile:

**Comment by ubruhin** [comment]
Hmm okay, strange...

Btw, I forgot to mention that there might be a "library~" directory in your project - if yes, that would be a backup of the whole "library" directory, created right before starting the project library update :)

**Comment by mdubinko** [comment]
This seems like potentially a duplicate issue. Demostrated herein. (unlisted video) 
https://youtu.be/TV7EBEiH7qc

Descripion: I created a new library component, a 16-segment LED display. I initially got the pin assignments wrong, so I went to the Packages screen and reassigned the pads. This resulted in things being broken all over the Devices screen. The 'Component Signal' settings all reverted to not connected, and when trying to reset them, I got an error message about the UUID not existing.

(To solve it, I had to manually edit my library files. Thank goodness for BBEdit rectangular selection. I bulk copied all the pad UUIDs from device.lp and blasted them into package.lp. This made the UUIDs exist, though the assignments were wrong. I was able to manually reassign pads to component signals in the UI, and now all is fine.)

**Comment by ubruhin** [comment]
@mdubinko I don't think your issue is a duplicate of this one. This issue is about the project's library, but yours is related to the workspace library. Is it possible that you removed and re-added pads in the package *after* creating the device? If yes, that's probably the problem. Pads must not be removed afterwards, as this breaks all devices using that package (the library editor should have shown a warning in that case).

> but I can't see the component to add it to a schematic

Probably the component doesn't have a category assigned? Only components with categories assigned are listed in the "add component" dialog.

**Comment by ubruhin** [comment]
As the whole file system access was refactored in #422, it's possible that this issue is now fixed automatically (although I could never reproduce it and thus I don't know what was going wrong exactly). 

I close this issue for now. If it still happens with the upcoming 0.1.1 release some time, we could re-open it.

**Comment by Scrappers-glitch** [comment]
Hello there !

I have downloaded librePCB 0.1.6 snap build.

And i am getting this when trying to download `LibrePCB Base v0.1.1` and `ST Microelectronics v0.2` from library manager which i assume it contains the basic circuit components, i am getting this error after downloading and during installing: 

```
Error while extracting the ZIP file 
"/home/pavl/LibrePCB-Workspace/v0.1/libraries/remote/a9ddf0c6-9b1c-4730-b300-01b4f192ad40.lplib.zip".
```

Now, when i try to pick a resistor and place it on my circuit, i am getting this error: 
```
The component with the UUID "ef80cd5e-2689-47ee-8888-31d04fc99174" does not exist in the workspace library!
```
Is there something to do ? 

Thank you !

**Comment by dbrgn** [comment]
@Scrappers-glitch does the downloading error for the library happen every time? You need the "LibrePCB Base" library for placing a resistor. You can try re-downloading it through the library manager.

**Comment by Scrappers-glitch** [comment]
> @Scrappers-glitch does the downloading error for the library happen every time? You need the "LibrePCB Base" library for placing a resistor. You can try re-downloading it through the library manager.

Yes, it happens every time, can I download it manually and link it to the library manager ?

**Comment by Scrappers-glitch** [comment]
> it's possible that this issue is now fixed automatically (although I could never reproduce it and thus I don't know what was going wrong exactly).

Something is wrong while extracting the zip of `a9ddf0c6-9b1c-4730-b300-01b4f192ad40.lplib.zip`, i can reproduce this issue, you should reopen it, i could help in fixing it, just let me know what to do....

I can find the folder for the base library as `.tmp` inside the workspace: 

![image](https://user-images.githubusercontent.com/60224159/180861950-290b1f0c-ab44-4324-aaa3-e71709382450.png)

**Comment by dbrgn** [comment]
I suspect something goes wrong when downloading, and then you get a 0-byte file that cannot be decompressed... @ubruhin is there some logging that could be used for debugging this?

**Comment by Scrappers-glitch** [comment]
And here is the list tree, i tried to edit the main directory and remove the `tmp` to make it readable but am getting a new error of `193ef70d-8dab-4a6c-a672-274c5bf09b68` component is missing when trying to add a new resistor, as you can see the component of resistor (`ef80cd5e-2689-47ee-8888-31d04fc99174`) does actually exists !
```
└── LibrePCB-Libraries-LibrePCB_Base.lplib-29faa39
    ├── cmp
    │   ├── 0c5052ab-eef9-45fa-9c52-bdd1aac1157d
    │   │   └── component.lp
    │   ├── 0ca0993c-bf55-4529-865b-849ef898b28a
    │   │   └── component.lp
    │   ├── 10bc0fd5-8703-4a9d-9694-0be7ad2c160f
    │   │   └── component.lp
    │   ├── 131de1ab-343f-461a-856d-1c812562d0a9
    │   │   └── component.lp
    │   ├── 1ee29522-c9ae-4119-98f5-4e4df386ec75
    │   │   └── component.lp
    │   ├── 1fc1a663-c393-41a9-b4b5-b2b3f4d6f8c8
    │   │   └── component.lp
    │   ├── 20f51f1d-ba6d-4319-961e-85645888661f
    │   │   └── component.lp
    │   ├── 23557088-3e20-4c09-9347-fa6be4f8fc92
    │   │   └── component.lp
    │   ├── 27bf51af-a016-45b4-87cb-844c72a38d66
    │   │   └── component.lp
    │   ├── 2b24b18d-bd95-4fb4-8fe6-bce1d020ead4
    │   │   └── component.lp
    │   ├── 2e833724-610d-44d8-bd1b-e291e0002736
    │   │   └── component.lp
    │   ├── 307aaecd-f053-4508-b41f-052a4799bcd5
    │   │   └── component.lp
    │   ├── 32889ae4-8521-4fdb-a5c9-f7d4a6402c36
    │   │   └── component.lp
    │   ├── 3373e04c-4305-4d8b-9963-750f786390cd
    │   │   └── component.lp
    │   ├── 403ed853-7f13-44b6-9325-6579c04f353a
    │   │   └── component.lp
    │   ├── 506bd124-6062-400e-9078-b38bd7e1aaee
    │   │   └── component.lp
    │   ├── 56f696e9-faef-4ec0-adf3-e64b134ada68
    │   │   └── component.lp
    │   ├── 58c3c6cd-11eb-4557-aa3f-d3e05874afde
    │   │   └── component.lp
    │   ├── 65343938-0df8-4c36-b9f8-ac405df27682
    │   │   └── component.lp
    │   ├── 6c9b08a8-dbf4-4db8-b0b2-b3cde89e43c4
    │   │   └── component.lp
    │   ├── 6eedad0b-5b41-4233-9b7b-8be1ee8527e0
    │   │   └── component.lp
    │   ├── 6f959627-00d2-4652-bfd9-f6ea0162ccdd
    │   │   └── component.lp
    │   ├── 721bf1fc-d360-4807-9aba-0cd1b5a228ad
    │   │   └── component.lp
    │   ├── 765e0ee6-e02d-4033-9fd5-f894bc788a84
    │   │   └── component.lp
    │   ├── 7deb8f5e-9ae3-4bfd-8142-14b7cad67488
    │   │   └── component.lp
    │   ├── 8076f6be-bfab-4fc1-9772-5d54465dd7e1
    │   │   └── component.lp
    │   ├── 82207562-d317-4911-96fa-ceb0f45fd806
    │   │   └── component.lp
    │   ├── 8a7b078b-ab3a-4c4c-ae2e-c6e2f717aa19
    │   │   └── component.lp
    │   ├── 984bb9b3-2d2c-44a0-8cac-478778182d78
    │   │   └── component.lp
    │   ├── 98e71eaa-1b90-4a84-8529-ed4f493e5c41
    │   │   └── component.lp
    │   ├── 9d043413-9574-4727-af3a-21c5623cffae
    │   │   └── component.lp
    │   ├── 9d3ef004-edfe-4ebc-b194-0da58242825d
    │   │   └── component.lp
    │   ├── a07aabe2-4127-40a9-9976-db1d6090c667
    │   │   └── component.lp
    │   ├── a4763181-79f0-4568-9212-bbc54e6f9aaa
    │   │   └── component.lp
    │   ├── a8e278ae-6e92-4713-b37b-a2c76c7066fb
    │   │   └── component.lp
    │   ├── b0cfc007-9834-43b4-bc7e-b348dd299326
    │   │   └── component.lp
    │   ├── b4f7c396-e6e7-407c-9e91-44996137c332
    │   │   └── component.lp
    │   ├── b847a54b-a83e-47ff-a976-7704bf05bbf4
    │   │   └── component.lp
    │   ├── b91cf23a-4f07-4b99-8f52-0b42304aef20
    │   │   └── component.lp
    │   ├── c54375c5-7149-4ded-95c5-7462f7301ee7
    │   │   └── component.lp
    │   ├── c93e340d-71f1-435c-a027-ceeac5b6dc1c
    │   │   └── component.lp
    │   ├── cdb73685-aaa1-483d-9143-6f7e40d1261a
    │   │   └── component.lp
    │   ├── cf77ee2e-3915-4a15-8bbb-5a8153c38a2a
    │   │   └── component.lp
    │   ├── d167e0e3-6a92-4b76-b013-77b9c230e5f1
    │   │   └── component.lp
    │   ├── d2ebf431-6989-4ccc-90d7-a33b558b6e61
    │   │   └── component.lp
    │   ├── e5732e00-3946-4c45-b5c0-11519326a87c
    │   │   └── component.lp
    │   ├── eb6bd73c-0104-44ed-9dff-f10f326a0291
    │   │   └── component.lp
    │   ├── ef80cd5e-2689-47ee-8888-31d04fc99174
    │   │   └── component.lp
    │   ├── f320ffc0-436c-4d1c-86d1-141d01c81295
    │   │   └── component.lp
    │   └── fa191b80-60d2-478d-a564-d677bd0fac6a
    │       └── component.lp
    ├── cmpcat
    │   ├── 1039f038-20a6-4bfe-89c1-99f34fbb45bd
    │   │   └── component_category.lp
    │   ├── 1f60e4f0-f86b-4f90-9081-187099ca03a7
    │   │   └── component_category.lp
    │   ├── 20580770-121f-422e-be2b-420c000487ed
    │   │   └── component_category.lp
    │   ├── 213bd44f-f375-41d8-8fdd-0652eb893e27
    │   │   └── component_category.lp
    │   ├── 22151601-c2d9-419a-87bc-266f9c7c3459
    │   │   └── component_category.lp
    │   ├── 267ee59f-c903-4391-bb98-6d7527dc7dd1
    │   │   └── component_category.lp
    │   ├── 2b70644c-2269-41f9-9381-6c9677330e01
    │   │   └── component_category.lp
    │   ├── 2bb3fe20-2ae6-4dfa-92f8-20004eaafe90
    │   │   └── component_category.lp
    │   ├── 2d036c7b-992d-44e5-8f69-f2d263163fea
    │   │   └── component_category.lp
    │   ├── 30643e64-b131-4eb5-b956-ceb3cb02413a
    │   │   └── component_category.lp
    │   ├── 38665d59-1fed-4cd0-8bcd-57af52fefb29
    │   │   └── component_category.lp
    │   ├── 38b3555a-bfea-4422-8d01-d4f2f94b1192
    │   │   └── component_category.lp
    │   ├── 39d4b3d6-4168-4fe2-9420-fbf7eb3ac248
    │   │   └── component_category.lp
    │   ├── 3ae7605e-f9e6-499f-bfa4-f72f62abd6f8
    │   │   └── component_category.lp
    │   ├── 3be18dae-a305-41ea-9e16-036b57cc21f8
    │   │   └── component_category.lp
    │   ├── 3dcfe933-b2ee-49fb-a66f-275bf51849bb
    │   │   └── component_category.lp
    │   ├── 3e181c86-f883-4d48-aa60-22db32b1b495
    │   │   └── component_category.lp
    │   ├── 4a2a74fb-e458-4903-9251-f0287a2fd04f
    │   │   └── component_category.lp
    │   ├── 4a4e3c72-94fb-45f9-a6d8-122d2af16fb1
    │   │   └── component_category.lp
    │   ├── 4b931698-8e30-4d59-98c8-6fcf452f4be2
    │   │   └── component_category.lp
    │   ├── 4d1c95f4-2e60-4f50-a15b-ba9186a01b1a
    │   │   └── component_category.lp
    │   ├── 5347fa0a-108b-4fa0-b97d-947d35c218be
    │   │   └── component_category.lp
    │   ├── 56f01b1d-0723-4c33-88f7-9968cf96b50f
    │   │   └── component_category.lp
    │   ├── 58601391-795b-4ebf-ae55-984352167ce4
    │   │   └── component_category.lp
    │   ├── 59112789-151e-4f46-b71b-ff33d023927a
    │   │   └── component_category.lp
    │   ├── 5a6e8219-34b8-44f2-a7c2-3a97ddf7456a
    │   │   └── component_category.lp
    │   ├── 651dca6f-67a5-4ee3-a07f-ed028243a6e2
    │   │   └── component_category.lp
    │   ├── 66367c6f-741a-4b3e-a7c6-af32a04b5b9b
    │   │   └── component_category.lp
    │   ├── 688c7dcd-2fcf-4655-90b6-777a42963cbe
    │   │   └── component_category.lp
    │   ├── 6b8cb737-30d9-49e2-831c-34aa909ef85e
    │   │   └── component_category.lp
    │   ├── 6dd43c27-b5f0-49fa-a8c0-926912d1dc35
    │   │   └── component_category.lp
    │   ├── 70421345-ae1d-4fed-aa60-e7619524b97f
    │   │   └── component_category.lp
    │   ├── 71b1d6fa-cc7c-44ea-82c3-4396dfd3d7cc
    │   │   └── component_category.lp
    │   ├── 7587f34f-a63d-45f1-8613-ef1e2375e240
    │   │   └── component_category.lp
    │   ├── 75a26b1f-c84e-4a3d-a20d-021d37e6697d
    │   │   └── component_category.lp
    │   ├── 7891f51d-8e92-4686-ab46-718ffb7f82db
    │   │   └── component_category.lp
    │   ├── 798db79a-f83e-49c5-bc4c-33e77c8501d6
    │   │   └── component_category.lp
    │   ├── 7c698c64-e0f4-4885-bc29-566c4828071e
    │   │   └── component_category.lp
    │   ├── 7da60b1f-96a7-4026-bd84-ab4a51883101
    │   │   └── component_category.lp
    │   ├── 8226f987-124b-4776-9084-3821dd51c272
    │   │   └── component_category.lp
    │   ├── 8625295c-d121-45c3-8bf9-32f5fce71722
    │   │   └── component_category.lp
    │   ├── 8c508cce-7904-44b9-a300-d9cc6eb44a6d
    │   │   └── component_category.lp
    │   ├── 8ca4f9fb-3dd3-4c1e-a097-6601b437bbc6
    │   │   └── component_category.lp
    │   ├── 8e4af7b5-cbff-4409-9698-f2b545218075
    │   │   └── component_category.lp
    │   ├── 8f9cbf8f-eada-4981-86a3-b789955a789a
    │   │   └── component_category.lp
    │   ├── 9066f53a-1270-466a-9b68-5501ffe41c08
    │   │   └── component_category.lp
    │   ├── 907a87d1-1050-46b0-95bd-f7c9e9068b77
    │   │   └── component_category.lp
    │   ├── 9109b7fd-e834-437f-b98d-114febc103e9
    │   │   └── component_category.lp
    │   ├── 9212a13b-d12d-4d13-b2c3-07a6c86e5e5c
    │   │   └── component_category.lp
    │   ├── 92fd938b-2eef-45be-b4e8-65b03a5dbfe3
    │   │   └── component_category.lp
    │   ├── 982705c8-9e88-4dab-89eb-fcea58afbeae
    │   │   └── component_category.lp
    │   ├── 9a25af45-d6a3-4c5a-af08-d68a148e9ca0
    │   │   └── component_category.lp
    │   ├── 9ce1e2fa-696e-405e-8c6a-c8fb0269fb5a
    │   │   └── component_category.lp
    │   ├── a159772a-3fc2-4037-ab59-f9d59b3ee6fc
    │   │   └── component_category.lp
    │   ├── a4c3b94b-5321-4adb-8a77-d3b53f178cee
    │   │   └── component_category.lp
    │   ├── a621d579-8715-4edf-816e-5715f97cd2d7
    │   │   └── component_category.lp
    │   ├── a6565eed-ad95-4be7-a29c-39259b1bea5f
    │   │   └── component_category.lp
    │   ├── a9aeef4d-41c2-440f-add5-cd1999e31e8f
    │   │   └── component_category.lp
    │   ├── ade6d8ff-3c4f-4dac-a939-cc540c87c280
    │   │   └── component_category.lp
    │   ├── adf8d941-787c-4b02-8772-fbc0a587463d
    │   │   └── component_category.lp
    │   ├── b1a06201-8efe-418d-ad1b-1e000cc12e63
    │   │   └── component_category.lp
    │   ├── b3adfa1e-b878-44f6-902a-14ef3dad7a14
    │   │   └── component_category.lp
    │   ├── b76b4ce6-4d44-4510-8e9d-688d44c697e5
    │   │   └── component_category.lp
    │   ├── b9fc7966-d5d2-4379-a029-5b65b69a4fbb
    │   │   └── component_category.lp
    │   ├── bd6346a1-5cbf-4b28-a62b-637d8e1a225d
    │   │   └── component_category.lp
    │   ├── bfe2b7c8-1aa0-49f1-9f27-e015d65af08e
    │   │   └── component_category.lp
    │   ├── c011cc6b-b762-498e-8494-d1994f3043cf
    │   │   └── component_category.lp
    │   ├── c227644e-1d75-442d-8cd8-185b1e4e64dd
    │   │   └── component_category.lp
    │   ├── c3a8348e-f7a9-4607-8fe7-253076498ae2
    │   │   └── component_category.lp
    │   ├── c3dfb625-e6e4-46c1-a1df-d14eeecfc965
    │   │   └── component_category.lp
    │   ├── c4c12c6e-abcd-4856-8445-e96c2737423e
    │   │   └── component_category.lp
    │   ├── c4ddfbc6-6543-499d-9e95-3091e049b07f
    │   │   └── component_category.lp
    │   ├── c6be465e-4086-4785-9437-8ec0a76689b3
    │   │   └── component_category.lp
    │   ├── cd70ec84-a440-496f-b5c9-ece75e81de64
    │   │   └── component_category.lp
    │   ├── cda9162b-fa94-4b99-abe0-00fc02c6f5c1
    │   │   └── component_category.lp
    │   ├── d0618c29-0436-42da-a388-fdadf7b23892
    │   │   └── component_category.lp
    │   ├── d1b6d2ca-061d-4c5c-81b2-cc7ee26942cb
    │   │   └── component_category.lp
    │   ├── db02d12f-b838-404a-9c04-2873c8c5a439
    │   │   └── component_category.lp
    │   ├── e05f53ad-1c05-42c1-9124-d0ed3e445a2d
    │   │   └── component_category.lp
    │   ├── e0702c57-f31f-497d-ae1e-e31c4017f3c0
    │   │   └── component_category.lp
    │   ├── e1f123ad-a959-474c-8b61-b5cdc92e817f
    │   │   └── component_category.lp
    │   ├── e29f0cb3-ef6d-4203-b854-d75150cbae0b
    │   │   └── component_category.lp
    │   ├── e4c9b084-7ee2-4310-9b5c-dc66c736a6e0
    │   │   └── component_category.lp
    │   ├── e6aea6d6-ac09-4e2d-862d-49a70e62613e
    │   │   └── component_category.lp
    │   ├── e9663545-80dd-4658-9357-d4ef62e55168
    │   │   └── component_category.lp
    │   ├── e96d0dda-3100-4e68-8c33-e2a5bc80428b
    │   │   └── component_category.lp
    │   ├── f4abe9d8-20b1-4818-a007-0f1b527e621d
    │   │   └── component_category.lp
    │   ├── f7f9bc34-7966-4af4-8cd4-4b8101e0823a
    │   │   └── component_category.lp
    │   ├── f81c3abf-dfbe-4464-b4d9-cc9824959f56
    │   │   └── component_category.lp
    │   ├── f87a2a14-4dd2-49f0-b93d-0c78ac50cf53
    │   │   └── component_category.lp
    │   ├── f9db4ef5-2220-462a-adff-deac8402ecf0
    │   │   └── component_category.lp
    │   └── ffe32a67-4ee5-4494-9936-c9e4c0b147bc
    │       └── component_category.lp
    ├── dev
    │   ├── 0107bccf-17c5-47ac-ae7a-75b057ba0a66
    │   │   └── device.lp
    │   ├── 0482b806-500c-4ebf-bad8-9bcdbe55dace
    │   │   └── device.lp
    │   ├── 06b68a18-b44e-4924-9856-d41c4e9d2506
    │   │   └── device.lp
    │   ├── 078650d3-483c-4b9e-a848-b14f1aad2edc
    │   │   └── device.lp
    │   ├── 0ae8ee60-3ae7-4c18-8c83-4fc5a95ba975
    │   │   └── device.lp
    │   ├── 0b1d13b7-23f9-4d14-9da3-2e4df1347682
    │   │   └── device.lp
    │   ├── 121c06df-1bfc-46fa-bda4-9cbcda821867
    │   │   └── device.lp
    │   ├── 168f4978-a6b6-4b5d-964d-be6733328650
    │   │   └── device.lp
    │   ├── 1b1ffd3f-9936-442a-aeec-2b5fb91973c7
    │   │   └── device.lp
    │   ├── 1e884376-8400-41a9-b87f-a427fd919033
    │   │   └── device.lp
    │   ├── 200606ec-3693-4681-89c3-18163177926c
    │   │   └── device.lp
    │   ├── 2127de78-83f7-478d-b76b-e24ce0ef8ca3
    │   │   └── device.lp
    │   ├── 22e1d311-ea1a-4abc-a0a1-34fa3877f0d4
    │   │   └── device.lp
    │   ├── 238f89b0-a52d-4e52-adf9-2faf595ae492
    │   │   └── device.lp
    │   ├── 2704f5e1-f8c2-4565-b8bd-5a7ea9946f08
    │   │   └── device.lp
    │   ├── 27c53d40-ec5f-42f7-b2f7-c766f77b2ba8
    │   │   └── device.lp
    │   ├── 2b823e2d-8295-48d8-a153-9dcdbca5d25d
    │   │   └── device.lp
    │   ├── 324de04a-f2ac-4134-b8bf-914173c150bf
    │   │   └── device.lp
    │   ├── 35fc5e10-cbf4-48db-91b3-1844aad77921
    │   │   └── device.lp
    │   ├── 3a5ce2ef-4c16-459f-a6b8-f914508a17a1
    │   │   └── device.lp
    │   ├── 3cf91223-569c-4cf5-b2ec-64fa283bd04f
    │   │   └── device.lp
    │   ├── 3d4b443a-8cfa-4940-bf17-eeb49c7b8a57
    │   │   └── device.lp
    │   ├── 4353b100-937f-4c82-be90-62f4b632ab8b
    │   │   └── device.lp
    │   ├── 46eba43f-55d2-4bab-bc17-1492e21c9741
    │   │   └── device.lp
    │   ├── 483a71eb-318e-448e-82ff-f02efc4821aa
    │   │   └── device.lp
    │   ├── 52876458-2396-4ade-8280-6a033efa911f
    │   │   └── device.lp
    │   ├── 5367ea13-ca04-4f05-8d90-7ee4f23e8c65
    │   │   └── device.lp
    │   ├── 5512ed40-604c-43fd-802c-1da10d419357
    │   │   └── device.lp
    │   ├── 5738d8f9-4101-4409-bd46-d9c173b40d60
    │   │   └── device.lp
    │   ├── 58678317-f386-416d-890a-470f59ab3fba
    │   │   └── device.lp
    │   ├── 5ff4a2e5-3593-49ab-bdbe-f9b68015a3ed
    │   │   └── device.lp
    │   ├── 61638d39-3132-48ac-9b13-5bc55f6fde54
    │   │   └── device.lp
    │   ├── 63175133-da03-43aa-9c34-4d3b968a19fe
    │   │   └── device.lp
    │   ├── 6a76fc34-d5cf-42c8-b17e-030110097721
    │   │   └── device.lp
    │   ├── 7186a6b2-4042-451f-a698-288394ec67a7
    │   │   └── device.lp
    │   ├── 781c74f2-a57e-4ada-96c9-1687d9ee65c0
    │   │   └── device.lp
    │   ├── 7ab2c481-4371-4064-a089-609596085821
    │   │   └── device.lp
    │   ├── 85f42ea9-d45b-48ed-8a77-73e6cdfd7491
    │   │   └── device.lp
    │   ├── 86b781d0-a0a0-4660-a525-c431f199ddbe
    │   │   └── device.lp
    │   ├── 88128b66-bb62-4964-ac98-a81e3a669940
    │   │   └── device.lp
    │   ├── 8b42ed9a-addf-4c67-a11b-531dcd0c6a17
    │   │   └── device.lp
    │   ├── 99351a67-0fa7-40b1-a034-56da111e056a
    │   │   └── device.lp
    │   ├── a0e021c0-90ab-4415-802e-40a847f682c8
    │   │   └── device.lp
    │   ├── a5153d16-a78b-439c-876b-12d12a607237
    │   │   └── device.lp
    │   ├── a6a6744d-7d3b-450a-b782-feca43939ca5
    │   │   └── device.lp
    │   ├── a8b774c7-d868-4138-93c9-88c0c80598a1
    │   │   └── device.lp
    │   ├── aa57914f-7a73-4393-9b7c-114174927469
    │   │   └── device.lp
    │   ├── aef46315-d199-4a31-90ca-0c41bcca3c2f
    │   │   └── device.lp
    │   ├── c139e505-592b-46ba-bdf2-acb7383ea0cd
    │   │   └── device.lp
    │   ├── c636f3fb-c815-422e-b212-c5703e6b84b7
    │   │   └── device.lp
    │   ├── c6ed996a-cd9a-4f47-9d0e-64d53d411b7a
    │   │   └── device.lp
    │   ├── c73ebe0d-a8eb-41ca-a1d4-aacc8d061afe
    │   │   └── device.lp
    │   ├── d0691a38-acb7-4a41-b889-99d7a2f92b10
    │   │   └── device.lp
    │   ├── d562e240-5f9a-497b-acd0-00c830abd7f4
    │   │   └── device.lp
    │   ├── dee6be59-7cc1-4bf8-935b-781d8ea2f154
    │   │   └── device.lp
    │   ├── e80a2ba5-13dd-41f2-9cdc-f29e30a9e2ed
    │   │   └── device.lp
    │   ├── f7fb22e8-0bbc-4f0f-aa89-596823b5bc3e
    │   │   └── device.lp
    │   ├── f83a5ae8-7f42-42be-9dd6-e762f4da2ec2
    │   │   └── device.lp
    │   └── fc557239-9f2e-4783-8777-0901c8db4e3c
    │       └── device.lp
    ├── library.lp
    ├── library.png
    ├── LICENSE.txt
    ├── pkg
    │   ├── 00e2b41b-22fb-4b98-8606-86c8ca0ae5ea
    │   │   └── package.lp
    │   ├── 01477c3a-e4f2-4704-8928-881b1c195ccf
    │   │   └── package.lp
    │   ├── 01808b65-9f74-407b-a70c-d82f238f04f2
    │   │   └── package.lp
    │   ├── 01915bf3-9987-4413-99cb-af4cd2862b73
    │   │   └── package.lp
    │   ├── 01b05306-c11a-4bd0-ab59-9841f195056d
    │   │   └── package.lp
    │   ├── 022bf744-7e07-4da6-8552-6b2bf628a075
    │   │   └── package.lp
    │   ├── 0243cef5-0c2a-4845-82d0-b9bf7a7a872f
    │   │   └── package.lp
    │   ├── 0355c703-3b47-4af6-b3fd-7767a62e1421
    │   │   └── package.lp
    │   ├── 03b21bab-a15f-4e22-8935-4d63a8862961
    │   │   └── package.lp
    │   ├── 03e31e01-0a1e-4dd0-9c4c-5edf057b0d94
    │   │   └── package.lp
    │   ├── 03fbc445-c49f-4059-93f4-1bfa61bbd570
    │   │   └── package.lp
    │   ├── 0402a5fb-33b0-41d1-9959-29cf1444ab19
    │   │   └── package.lp
    │   ├── 0444d2ac-d1fc-48a3-8c31-a9e56c0e7ce0
    │   │   └── package.lp
    │   └── 046b35fd-1e95-4475-9b89-2a80a48f1d1e
    └── README.md

220 directories, 218 files
```

The resistor file has got this, i have no idea if this is corrupted format ? or not but seems fine for me...
```
(librepcb_component ef80cd5e-2689-47ee-8888-31d04fc99174
 (name "Resistor")
 (name (locale "de_DE") "Widerstand")
 (name (locale "fr_FR") "Résistance")
 (description "Ordinary fixed resistor")
 (description (locale "de_DE") "Gewöhnlicher Festwiderstand")
 (keywords "resistor,resistance")
 (author "U. Bruhin")
 (version "0.2")
 (created 2015-03-01T20:00:00Z)
 (deprecated false)
 (category 1039f038-20a6-4bfe-89c1-99f34fbb45bd)
 (schematic_only false)
 (default_value "{{RESISTANCE}}")
 (prefix "R")
 (attribute "RESISTANCE" (type resistance) (unit ohm) (value ""))
 (signal 3452d36e-1ce8-4b7c-8e5b-90c2e4929ed8 (name "1") (role passive)
  (required true) (negated false) (clock false) (forced_net "")
 )
 (signal ad623f98-9e73-49c3-9404-f7cfa99d17cd (name "2") (role passive)
  (required true) (negated false) (clock false) (forced_net "")
 )
 (variant a5995314-f535-45d4-8bd8-2d0b8a0dc42a (norm "IEC 60617")
  (name "European")
  (description "")
  (gate 5988c66a-f507-487e-9b40-486502e8c9bd
   (symbol 75372c18-3ba4-42e8-b3b2-2eb5039d441e)
   (position 0.0 0.0) (rotation 0.0) (required true) (suffix "")
   (pin 2731f213-ee20-4eb0-a18b-ad97d9fea7a0 (signal ad623f98-9e73-49c3-9404-f7cfa99d17cd) (text none))
   (pin 2e4f8b10-ad5a-443e-b408-e183efaa4758 (signal 3452d36e-1ce8-4b7c-8e5b-90c2e4929ed8) (text none))
  )
 )
 (variant d16e1f44-16af-4773-a310-de370f744548 (norm "IEEE 315")
  (name "American")
  (description "")
  (gate 169660fd-968a-4d3e-83f5-47f973b4ecd8
   (symbol 193ef70d-8dab-4a6c-a672-274c5bf09b68)
   (position 0.0 0.0) (rotation 0.0) (required true) (suffix "")
   (pin 2b3dd7f8-043b-4d43-9302-9300ba356de7 (signal ad623f98-9e73-49c3-9404-f7cfa99d17cd) (text none))
   (pin f42020e8-c53f-4ff2-947e-07879cf42546 (signal 3452d36e-1ce8-4b7c-8e5b-90c2e4929ed8) (text none))
  )
 )
)
```

**Comment by Scrappers-glitch** [comment]
Alright @dbrgn , problem solved !
I wasn't having enough space on my /home, lol....so may be a suggestion is to produce a better message to the user, may be !

Thank you !

**Comment by dbrgn** [comment]
Hehe, that explains it 🙂

Yes, better diagnostics for library downloading and decompressing would be good.
